receiver/apachesparkreceiver/config.go

```diff
@@ -24,6 +26,7 @@ var (
 // Config defines the configuration for the various elements of the receiver agent.
 type Config struct {
 	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+	metadata.MetricsBuilderConfig           `mapstructure:",squash"`
 	confighttp.HTTPClientSettings           `mapstructure:",squash"`
 	WhitelistedApplicationNames             []string `mapstructure:"whitelisted_application_names"`
```

Comment:
```suggestion
	Applications             []string `mapstructure:"applications"`
```

Reply:
I'm thinking we keep this as application_names / ApplicationNames, just since applications also have IDs and I'd like to be as clear as possible. Also, multiple Applications can have the same name, but never the same ID, and I just wonder if "Applications" doesn't make the functionality quite as clear in that regard. Thoughts?

Reply:
> multiple Applications can have the same name

If that's the case, should we require the user to specify the ID?

Reply:
Generally though, I'm fine with keeping `_names` if we think it is not implied.

Reply:
That was actually the original implementation, but we swapped since the IDs in question aren't pre-defined by the user like the application name is - they're autogenerated based on the timestamp when the application was run. i.e., a Spark app I ran this morning had an ID of `local-1684767537244`.